ibis/expr/operations/core.py

```diff
@@ -75,11 +75,17 @@ def __coerce__(
         if isinstance(value, Value):
             return value
 
-        try:
+        if T is dt.Integer:
```

Comment:
Special branches for numeric type families where we want to coerce to the smallest numeric types.